tests/namespace/namespace_test.go

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package namespace
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/kubearmor/KubeArmor/tests/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = BeforeSuite(func() {
+	err := K8sApply([]string{"res/ubuntu-deployment.yaml"})
+	Expect(err).To(BeNil())
+	KspDeleteAll()
+})
+
+var _ = AfterSuite(func() {
+	err := K8sDelete([]string{"res/ubuntu-deployment.yaml"})
+	Expect(err).To(BeNil())
+	KspDeleteAll()
+})
+
+func getUbuntuPod(name string, ant string) string {
+        pods, err := K8sGetPods(name, "nptest", []string{ant}, 60)
+        Expect(err).To(BeNil())
+        Expect(len(pods)).To(Equal(1))
+        return pods[0]
+}
+
+
+var _ = Describe("Namespace", func() {
+	var ubuntu string
+
+	BeforeEach(func() {
+		ubuntu = getUbuntuPod("ubuntu-1-deployment-","container.apparmor.security.beta.kubernetes.io/ubuntu-1-container: localhost/kubearmor-nptest-ubuntu-1-deployment-ubuntu-1-container")
+	})
+
+	AfterEach(func() {
+		KarmorLogStop()
+		KspDeleteAll()
+	})
+
+	Describe("Policy Apply", func() {
+
+		It("check default posture for network", func() {
+			// Apply policy
+			err := K8sApply([]string{"res/network.yaml"})
+			Expect(err).To(BeNil())
+
+			// Start Kubearmor Logs
+			err = KarmorLogStart("policy", "nptest", "Network", ubuntu)
+			Expect(err).To(BeNil())
+
+			sout, _, err := K8sExecInPod(ubuntu, "nptest",[]string{"bash", "-c", "curl google.com"})
+			Expect(err).To(BeNil())
+			fmt.Printf("OUTPUT: %s\n", sout)
+			Expect(sout).To(MatchRegexp("Could not resolve host: google.com"))
+
+			// check policy violation alert
+			_, alerts, err := KarmorGetLogs(5*time.Second, 25)
+			Expect(err).To(BeNil())
+			Expect(len(alerts)).To(BeNumerically(">=", 1))
+			fmt.Printf("%+v\n", alerts[0])
+			Expect(alerts[len(alerts)-1].PolicyName).To(Equal("DefaultPosture"))
+			Expect(alerts[len(alerts)-1].Action).To(Equal("Block"))
+			Expect(alerts[len(alerts)-1].ProcessName).To(Equal("/usr/bin/curl"))
```

Comment:
Using `[len(alerts)-1]` is tricky. This way of handling alerts is a recipe for flaky CI.